### PR TITLE
Added autoleave feature

### LINKED ISSUE #86: Add auto exit feature
Currently, we only have the _auto-join_ feature and not the _auto-exit_ feature.
We can add this feature and can improve the user experience.

### PR DESCRIPTION
**Pull request by XanderWatson**
Closes #86

### CODE REVIEW

File: src/pages/Popup/scripts/alarm.js [hunk 1]
```diff
@@ -116,18 +120,38 @@ async function AddAlarm_click(course, dates) {
...
+  if (alarm.EndTime != '' && autoleaveswitch == true) {
+    endtimecheck = true;
+  }
+  var values = {
+    Authuser: authuser,
+    BeforeMinutes: beforeminutes,
+    BeforeSeconds: beforeseconds,
+    AutoJoin: autojoin,
+    AutoLeaveSwitch: autoleaveswitch,
+    EndTimeCheck: endtimecheck
```
Comment (DarshitJain04) [inline src/pages/Popup/scripts/alarm.js:141]: I realized that setting `EndTimeCheck` in storage and using the stored value whenever an alarm is triggered can lead to a big issue. 
Say I set an alarm (with no end time) for today at 8 pm. So in this case, `EndTimeCheck` won't be true and the old value would be stored in storage (the old value can be true or false)
Now, I set a new alarm (with end time) for today at 9 pm. In this case, `EndTimeCheck` will be true and we are saving that in storage.
When the 8 pm alarm gets triggered, we use the default values from storage. According to the values, `EndTimeCheck` would be true, but  in this case, we did not set the end time value, so it should have been false

So what we should do is, check `alarm.EndTime != ''` and  `autoleaveswitch == true` in background.js when the alarm is triggered. If it satisfies both the condition, we pass true, else false.

File: src/scripts/meet.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 const autoJoin = () => {
-  var initiateAutoJoin = setInterval(() => {
+  var initiateAutoJoin = setTimeout(() => {
```
Comment (DarshitJain04) [inline src/scripts/meet.js:2]: Any reason to change `setInterval` to `setTimeout` ?

**Review by DarshitJain04 — CHANGES_REQUESTED** [review]

File: src/pages/Popup/components/AlarmUp.jsx [hunk 3]
```diff
@@ -142,7 +142,7 @@ class Alarmview extends Component {
       link = await get_meetlink(data.A);
     }
     let details = await getDataFromStorage('Defaults');
-    let tab = await createTab(link, details.Authuser, details.AutoJoin);
+    let tab = await createTab(link, details.Authuser, details.AutoJoin, details.EndTimeCheck, data.EndTime);
```
[the remark this answers is not in the document]
Comment (XanderWatson) [reply src/pages/Popup/components/AlarmUp.jsx:145]: The 'data' variable in this file is actually equivalent to 'data.course' in background.js if you look closely.

File: src/pages/Popup/scripts/alarm.js (hunk 1 above)
Comment (XanderWatson) [reply src/pages/Popup/scripts/alarm.js:141]: Hmm makes sense. I'll change it.

File: src/scripts/meet.js (hunk 2 above)
Comment (XanderWatson) [reply src/scripts/meet.js:2]: Not really. Just didn't find a reason as to why was it `setInterval` in the first place.

Comment (DarshitJain04) [reply src/scripts/meet.js:2]: Reset the changes in this file.

Comment (XanderWatson) [reply src/scripts/meet.js:2]: Oh okay I get it now. `setInterval` was there incase it doesn't join in the first attempt. My bad.

**Comment by DarshitJain04** [comment]
Pull changes from master branch

**Review by DarshitJain04 — CHANGES_REQUESTED** [review]

File: src/pages/Popup/components/CustomForm.jsx [hunk 4]
```diff
@@ -95,6 +97,21 @@ class AddAlarmForm extends Component {
     this.check();
   };
 
+  handleEndTimeChange = async (event) => {
+    var val = event.target.value;
+    var d = new Date(val);
+    var error = false;
+    console.log(val, d);
+    if (val === '' || d < new Date()) {
```
Comment (DarshitJain04) [inline src/pages/Popup/components/CustomForm.jsx:105]: Add a check if the end time is greater than the start time or not. Currently, I can enter an end time which is before the start time. Have a look:
autoleave

**Review by DarshitJain04 — CHANGES_REQUESTED** [review]

**Review by DarshitJain04 — APPROVED** [review]

**Comment by DarshitJain04** [comment]
@XanderWatson we are good to go now 🚀 
Good job! 🎉

**Comment by XanderWatson** [comment]
> @XanderWatson we are good to go now 🚀 
> Good job! 🎉 

Yay! 🎉🥳

**Pull request merged**